src/app/core/learning-object-module/bundling/bundling.service.ts

```diff
@@ -39,4 +50,49 @@ export class BundlingService {
       return throwError(error);
     }
   }
+
+  toggleBundle(
+    learningObjectId: string,
+    fileIDs: string[],
+    state: boolean
+  ) {
+    const route = BUNDLING_ROUTES.TOGGLE_BUNDLE_FILE({ learningObjectId });
+
+    return this.http
+      .patch(
+        route,
+        {
+          fileIDs: fileIDs,
+          packagable: state
+        },
+        { headers: this.headers, withCredentials: true }
+      )
+      .pipe(
+
+        catchError(this.handleError)
+      )
+      .toPromise();
+  }
+
+
+  toggleDownloadModal(val?: boolean) {
+    this.showDownloadModal = val;
+  }
+
+  download(learningObjectId: string) {
+    this.toggleDownloadModal(true);
+    this.libraryService.downloadBundle(BUNDLING_ROUTES.DOWNLOAD_BUNDLE(learningObjectId));
+  }
+
+
+  downloadObject(event: MouseEvent, object: LearningObject, index: number) {
```

Comment:
This looks like the function from the component, this should stay in the component and the reference to this.libraryService.downloadBundle should be this.bundlingService.downloadBundle and the downloadBundle function from library.service.ts should be moved here, then this should work.